autovalet_navigation/src/move_base_listener.cpp

```diff
@@ -0,0 +1,204 @@
+#include <move_base_listener.h>
+#include <geometry_msgs/TransformStamped.h>
+
+
+void getCurrentPose(geometry_msgs::PoseStamped &pose)
+{
+    // get the current transform from map to base_link and store that in "pose"
+    geometry_msgs::TransformStamped transform;
+    tf2_ros::Buffer tf_buffer;
+    tf2_ros::TransformListener tf_listener(tf_buffer);
+    transform = tf_buffer.lookupTransform("icp_odom","map",ros::Time(0), ros::Duration(1.0));
```

Comment:
Check if there are any warnings due to ros::Time(0)? Sometimes changing that (or ros:Duration(1.0)) would ensure that we don't see those error/warning.

Reply:
time(0) means look up the transform now. duration(1) is the timeout for the lookup. i didn't see any errors. did you?